### PR TITLE
[GEOS-11134] Review assembly html files

### PR DESCRIPTION
**Pull request by jodygarnett**
[GEOS-11134](https://osgeo-org.atlassian.net/browse/GEOS-11134) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geoserver&utm_campaign=badge_info)

Required:

* [x] bin and war licenses/ folder does not include the full GPL.html license as expected
* [x] extensions do not always include README / LICENSE notice
* [x] RUNNING.html can link to https://docs.geoserver.org/latest/en/user/installation 

Nice to have:

* [ ] bin the logging/ and resources/ folders can be removed if the are not being used?
* [ ] javadoc is the frame presentation still available? 

To test locally:

``` bash
cd src
mvn clean install -Prelease
mvn assembly:single -nsu -N
```

The release/README.md now has these instructions also. Since someone on gitter resorted to look at the GitHub workflows to figure it out...

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by mprins** [comment]
Build seems to fail on a missing file
`Error: 7,444 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.2.0:single (default-cli) on project geoserver: Failed to create assembly: Error adding file to archive: /home/runner/work/geoserver/geoserver/src/release/target/html/extensions/h2/h2-license.html`

File: LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,410 @@
+# GeoServer License Notice
+
+GeoServer is distributed under the GNU General Public License Version 2.0 license:
+
+    GeoServer, open geospatial information server
+    Copyright (C) 2014-2020 Open Source Geospatial Foundation.
```
Comment (mprins) [inline LICENSE.md:6]:
```suggestion
    Copyright (C) 2014-2023 Open Source Geospatial Foundation.
```

Or perhaps no endyear

File: src/release/extensions/dxf/dxf-LICENSE.md [hunk 2]
```diff
@@ -15,7 +18,7 @@ Mauro Bartolomeoli, GeoSolutions Sas
     along with this program; if not, write to the Free Software
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 
-The full GPL license is available in this directory at GPL.txt
+This GeoServer plugin is licensed under the GNU General Public License (GPL). For more information see license [notice](licenses/NOTICE.md) and the full [GPL](licenses/GPL.html).
 
 This software is an extension for GeoServer. You can find full information
-on it at http://www.geoserver. 
+on it at https://www.geoserver.
```
Comment (mprins) [inline src/release/extensions/dxf/dxf-LICENSE.md:24]:
```suggestion
on it at https://www.geoserver.org
```

**Review by mprins — COMMENTED** [review]
Found some small things while skimming through, on mobile

**Comment by jodygarnett** [comment]
> Build seems to fail on a missing file `Error: 7,444 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.2.0:single (default-cli) on project geoserver: Failed to create assembly: Error adding file to archive: /home/runner/work/geoserver/geoserver/src/release/target/html/extensions/h2/h2-license.html`

This will be case sensitive file system problem!

File: LICENSE.md (hunk 1 above)
Comment (jodygarnett) [reply LICENSE.md:6]: I can process these files to include current year like I do for the header and footer.

**Comment by jodygarnett** [comment]
This has been having some heap space issues with building war; not sure if we should increase the limit. Re-running eventually passes tests.

**Pull request merged**